### PR TITLE
Added Docker build instructions

### PR DESCRIPTION
**Pull request by jraviotta**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -8,3 +8,4 @@ dist
 coverage
 .vscode
 config/jest/globalConfig.json
+database
```
Comment (Shawn8901) [inline .gitignore]: add a new line here

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    && apk del build-dependencies \
+    && rm -rf /var/cache/apk/*
+
+# Add app source from host
+COPY . /app
+
+# Expose a port
+EXPOSE 9999
+
+ENTRYPOINT ["tail", "-f", "/dev/null"]
```
Comment (Shawn8901) [inline Dockerfile]: same here

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+  mongodb:
+    image: mongo:latest
+    volumes:
+      - database:/data/db
+    command: mongod --smallfiles --bind_ip=0.0.0.0 --logpath=/dev/null
+    expose:
+      - 27017
+
+# Remove below comments to use this container. "adminMongo is a Web based user interface (GUI) to handle all your MongoDB connections/databases needs."
```
Comment (Shawn8901) [inline docker-compose.yml]: I think, you could remove that comment?
Maybe just hint, that this is the admin connection and can be commented out with `#` if not needed.

**Review by Shawn8901 — COMMENTED** [review]
I like the general idea of the make file!
Could you please add a short documentation of that (which would be the most common commands).

so `make build`
`make time-sync`
`make shell`
`make up`
`make start`
`make stop`
`make logs`

should fine a place in Readme.md

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,41 @@
+FROM node:10-alpine
+
+
+# Install all build dependencies
+# Add bash for debugging purposes
+RUN apk update && \
+    apk add --virtual build-dependencies \
+    build-base \
+    dos2unix \
+    python3-dev && \
```
Comment (Shawn8901) [inline Dockerfile]: i think python2 is a better dependency here?

File: .dockerignore [hunk 5]
```diff
@@ -0,0 +1,13 @@
+# .dockerignore everything
+**
+
+# except
+!/assets/*
+!/config/*
+!/legacy/*
+!/src/*
+!/jest.config.js
```
Comment (Shawn8901) [inline .dockerignore]: if you are looking for a productive image, there is no need for jest (its the unit-testing framework). i think you could remove it.

File: .dockerignore [hunk 6]
```diff
@@ -0,0 +1,13 @@
+# .dockerignore everything
+**
+
+# except
+!/assets/*
+!/config/*
+!/legacy/*
```
Comment (Shawn8901) [inline .dockerignore]: this could be removed, since we are not even compiling it (see tsconfig.json)

File: Dockerfile [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+    apk add bash
+
+# Add mainfest from host
+WORKDIR /app
+COPY yarn.lock /app/
+
```
Comment (Shawn8901) [inline Dockerfile]: if you add `COPY package.json /app/`here, yarn is able to build on Line 30.
Also include  `COPY tsconfig.json /app/` and `COPY src /app/src`

Comment (Shawn8901) [inline Dockerfile]: We also should add here a mount for the config file. otherwise tsc will fail.

**Review by Shawn8901 — DISMISSED** [review]

**Comment by jraviotta** [comment]
It appears that the container builds now. Executing `yarn run start:dev` in the container shows signs of life but mongo refuses the connection. Probably need some settings in the config. Unfortunately, there doesn't seem to be an easy cross-platform dev solution. Bind mounts do not work on my Windows machine using WSL, so a workaround that runs on all the platforms will be troublesome.  Devs will either need a different `docker-compose.yml` or will need to edit the `server:volumes` key manually.

**Comment by Shawn8901** [comment]
I think we are here at a conceptional problem. 
The Configuration file has to be present at compile time. 
So we have a question to clarify:
* Should a user edit the original TypeScript-Configuration file?
* Or should the user edit the compiles config file?
* Or is the current config file concept really the right one or switch to something else, like json file which will be loaded at runtime?

for the first we will have to ship a docker container which is able to compile, and the user will have to compile it everything he wants to change a config parameter.

edit:
about your connection issue to mongodb: you have to change the hostname in `conf.ts` from `localhost` to `mongodb`.

**Comment by jraviotta** [comment]
I can confirm that changing db config to `mongodb` from `localhost` allows the app to boot and the database is visible in adminmongo at `localhost:1234`. Executing `yarn run start:dev` in the container returns `Error: gdax missing required credentials. Requires: apiKey, secret, password` which proves that the app is alive.

This PR can be merged with the understanding that automated image builds will fail until we decide how to deal with the db connection in `conf-sample.ts`. Of course that won't be an issue until Travis.ci is told to build images anyway. Users who want to build docker images can do so by setting db config in their local `conf.ts` file and building the image locally. The only local dependency is Docker.

**Review by notVitaliy — APPROVED** [review]
This looks very promising. I'm going to mark this as approved and merge it in.

Feel free to create a new PR if you want to make some changes. Trying to keep PR sizes small and incremental. Less mental overhead to do code reviews that way.

**Pull request merged**